4 polish writing smart contracts and session code pages

jonas089 opened the pull request:
Combined "session-code" and "contracts-vs-session-code" and added a "Vault" example.
Compiled successfully and no issues rendering the page:
<img width="568" alt="Screenshot 2023-01-31 at 18 59 57" src="https://user-images.githubusercontent.com/49498646/215844549-d13bcc78-97ea-44c8-ae14-8bab81f11896.png">

To Karol: Please merge this with any changes you have done in respect to Smart Contract testing. 
I consider this section as mostly complete and will move on for now.

Commit:
merge from old smart-contracts branch

Commit:
fix reflinks

Commit:
formatting

Commit:
remove multisig from doc, as it belongs in How-tos

Commit:
ready for merge

jonas089 commented:
Apparently a build check failed, but I don't know why as it worked perfectly fine on my machine. Please let me know if there is an issue that's to be resolved.

Commit:
upgrade docusaurus.config.js for latest version

ghost commented:
@jonas089 I see that you committed many additional changes, that are unrelated with task (content polishing):

1. :exclamation: Updated docusaurus dependency (with npm, which inadvertently deleted yarn.lock and created package-lock.json)
2. Modified git pre-commit hook
3. Updated docuzaurus config (hiding sidebar?)

**Note**: I expect that build failed, because of the first one.

Of course we are allowed to update docuzaurus, its config, hooks etc., however each change should be a separate task and separate PR. That way we separate concerns and make review process straightforward.

jonas089 commented:
> @jonas089 I see that you committed many additional changes, that are unrelated with task (content polishing):
> 
> 1. :exclamation: Updated docusaurus dependency (with npm, which inadvertently deleted yarn.lock and created package-lock.json)
> 2. Modified git pre-commit hook
> 3. Updated docuzaurus config (hiding sidebar?)
> 
> **Note**: I expect that build failed, because of the first one.
> 
> Of course we are allowed to update docuzaurus, its config, hooks etc., however each change should be a separate task and separate PR. That way we separate concerns and make review process straightforward.

Thanks, I've learned from the mistakes I made yesterday and I think I'm used to the workflow now 😃 took me a few hours to understand yarn as I've only used npm before.
